Source/Scene/ModelExperimental/ModelExperimentalDrawCommand.js

```diff
@@ -0,0 +1,489 @@
+  const commandList = getAllCommands(drawCommand);
+  const commandLength = commandList.length;
+
+  for (let i = 0; i < commandLength; i++) {
+    const command = commandList[i];
+    const renderState = clone(command.renderState, true);
+    renderState.cull.face = cullFace;
+    command.renderState = RenderState.fromCache(renderState);
+  }
+}
+
+function updateShowBoundingVolume(drawCommand) {
+  const debugShowBoundingVolume = drawCommand.debugShowBoundingVolume;
+
+  const commandList = getAllCommands(drawCommand);
+  const commandLength = commandList.length;
+  for (let i = 0; i < commandLength; i++) {
+    const command = commandList[i];
+    command.debugShowBoundingVolume = debugShowBoundingVolume;
+  }
+}
+
+/**
+ * Returns an array of the draw commands necessary to render the primitive.
+ *
+ * @param {FrameState} frameState The frame state.
+ *
+ * @returns {DrawCommand[]} The draw commands.
+ *
+ * @private
+ */
+ModelExperimentalDrawCommand.prototype.getCommands = function (frameState) {
+  const commandList = this._commandList;
+  const commandList2D = this._commandList2D;
+
+  const use2D = use2DCommands(this, frameState);
+
+  if (use2D && commandList2D.length === 0) {
+    const length = commandList.length;
+    for (let i = 0; i < length; i++) {
+      const command2D = derive2DCommand(commandList[i]);
+      commandList2D.push(command2D);
+    }
+
+    this._modelMatrix2DDirty = true;
+  }
+
+  if (this._modelMatrix2DDirty) {
+    updateModelMatrix2D(this, frameState);
+    this._modelMatrix2DDirty = false;
+  }
+
+  const commands = [];
+  commands.push.apply(commands, commandList);
+
+  if (use2D) {
+    commands.push.apply(commands, commandList2D);
+  }
+
+  return commands;
+};
+
+/**
+ * @private
+ */
+function deriveTranslucentCommand(command) {
+  const derivedCommand = DrawCommand.shallowClone(command);
+  derivedCommand.pass = Pass.TRANSLUCENT;
+  const rs = clone(command.renderState, true);
+  rs.cull.enabled = false;
+  rs.depthTest.enabled = true;
+  rs.depthMask = false;
+  rs.blending = BlendingState.ALPHA_BLEND;
+  derivedCommand.renderState = RenderState.fromCache(rs);
+
+  return derivedCommand;
+}
+
+/**
+ * If the model crosses the IDL in 2D, it will be drawn in one viewport but get
+ * clipped by the other viewport. We create a second command that translates
+ * the model matrix to the opposite side of the map so the part that was clipped
+ * in one viewport is drawn in the other.
+ *
+ * @param {DrawCommand} command The original draw command.
+ *
+ * @returns {DrawCommand} The derived command for rendering across the IDL in 2D.
+ *
+ * @private
+ */
+function derive2DCommand(command) {
+  const derivedCommand = DrawCommand.shallowClone(command);
+
+  // These will be computed in updateModelMatrix2D()
+  derivedCommand.modelMatrix = new Matrix4();
+  derivedCommand.boundingVolume = new BoundingSphere();
+
+  return derivedCommand;
+}
+
+function use2DCommands(drawCommand, frameState) {
```

Comment:
maybe name this `shouldUse2DCommands()` to make it more obvious that it returns a bool?